ext/standard/array.c

```diff
@@ -2057,13 +2055,25 @@ PHP_FUNCTION(array_fill_keys)
 }
 /* }}} */
 
-#define RANGE_CHECK_INIT_ARRAY(start, end) do { \
+#define RANGE_CHECK_DOUBLE_INIT_ARRAY(start, end) do { \
 		double __calc_size = ((start - end) / step) + 1; \
-		if (fabs(__calc_size) >= (double)HT_MAX_SIZE) { \
-			php_error_docref(NULL, E_WARNING, "The supplied range exceeds the maximum array size: start=%0.0f end=%0.0f", start > end ? end : start, start > end ? start : end); \
+		if (__calc_size > (double)HT_MAX_SIZE) { \
```

Comment:
`start < end` never occurs, and if it did, the size calculation would be incorrect. This is because `((start - end) / step)` (= `n`) would yield a negative, and then `+ 1` would make the size `n - 1`, rather than `n + 1` (for an inclusive range). If we look at where the macro is invoked, we can see that these checks (to ensure that `start > end`) are already performed:
http://lxr.php.net/xref/PHP_7_0/ext/standard/array.c#2185 (the enclosing IF ensures low > high)
http://lxr.php.net/xref/PHP_7_0/ext/standard/array.c#2198 (the enclosing IF ensures high > low)

Regarding the size check against `HT_MAX_SIZE`, can the maximum size of an array be `HT_MAX_SIZE - 1` then? If so, I'll update the PR to reflect that.